pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -157,8 +157,8 @@ def on_load_checkpoint(self, callback_state: Dict[str, Any]) -> None:
         self.patience = callback_state['patience']
 
     def _should_skip_check(self, trainer) -> bool:
-        from pytorch_lightning.trainer.states import TrainerState
-        return trainer.state != TrainerState.FITTING or trainer.sanity_checking
+        from pytorch_lightning.trainer.states import TrainerFn
```

Comment:
why the import inside the function instead of in the top ?

Reply:
Circular imports. To fix it properly, we would have to move the `states.py` file outside the trainer directory.

Reply:
does the trainer still depend on early stopping after https://github.com/PyTorchLightning/pytorch-lightning/pull/7069 ?

Reply:
Indirectly through general imports, see the error trace all over the place if I remove this local import in particular:

```python
test_early_stopping.py:25: in <module>
    from pytorch_lightning import seed_everything, Trainer
../../pytorch_lightning/__init__.py:21: in <module>
    from pytorch_lightning.callbacks import Callback  # noqa: E402
../../pytorch_lightning/callbacks/__init__.py:15: in <module>
    from pytorch_lightning.callbacks.early_stopping import EarlyStopping
../../pytorch_lightning/callbacks/early_stopping.py:28: in <module>
    from pytorch_lightning.trainer.states import TrainerFn
../../pytorch_lightning/trainer/__init__.py:18: in <module>
    from pytorch_lightning.trainer.trainer import Trainer
../../pytorch_lightning/trainer/trainer.py:25: in <module>
    from pytorch_lightning.accelerators import Accelerator
../../pytorch_lightning/accelerators/__init__.py:13: in <module>
    from pytorch_lightning.accelerators.accelerator import Accelerator  # noqa F401
../../pytorch_lightning/accelerators/accelerator.py:25: in <module>
    from pytorch_lightning.plugins.precision import ApexMixedPrecisionPlugin, NativeMixedPrecisionPlugin, PrecisionPlugin
../../pytorch_lightning/plugins/__init__.py:2: in <module>
    from pytorch_lightning.plugins.plugins_registry import (  # noqa: F401
../../pytorch_lightning/plugins/plugins_registry.py:21: in <module>
    from pytorch_lightning.plugins.training_type.training_type_plugin import TrainingTypePlugin
../../pytorch_lightning/plugins/training_type/__init__.py:4: in <module>
    from pytorch_lightning.plugins.training_type.deepspeed import DeepSpeedPlugin  # noqa: F401
../../pytorch_lightning/plugins/training_type/deepspeed.py:25: in <module>
    from pytorch_lightning.callbacks import GradientAccumulationScheduler
E   ImportError: cannot import name 'GradientAccumulationScheduler' from partially initialized module 'pytorch_lightning.callbacks' (most likely due to a circular import) (/Users/carlosmocholi/git/pytorch-lightning/pytorch_lightning/callbacks/__init__.py)
```